corehq/apps/accounting/models.py

```diff
@@ -500,13 +507,24 @@ def _get_account_by_created_by_domain(cls, domain):
     def get_enterprise_restricted_signup_accounts(cls):
         return BillingAccount.objects.filter(is_customer_billing_account=True, restrict_signup=True)
 
+    @classmethod
+    def get_enterprise_permissions_domains(cls, domain):
+        account = cls.get_account_by_domain(domain)
+        if account and account.permissions_source_domain:
+            return list(set(account.get_domains()) - set(account.permissions_ignore_domains))
```

Comment:
Shouldn't this return an empty list if `domain` is in the ignored list?

Reply:
resolved by the switch to `EnterprisePermissions`